caffe2: Fix lint errors in native/TensorAdvancedIndexing.cpp

EricGriffith opened the pull request:
Summary: See title

Test Plan: Sandcastle

Differential Revision: D72274536

malfet commented on aten/src/ATen/native/TensorAdvancedIndexing.cpp:
```diff
@@ -647,14 +647,14 @@ static Tensor reshape_indexer(
+  int64_t new_dims_before = 0, new_dims_after = 0, dims_indexed = 0;
   IntArrayRef replacement_shape;
-  for (const auto dim : c10::irange(indices_list.size())) {
+  for (const auto dim : c10::irange(static_cast<int64_t>(indices_list.size()))) {
```
Why this static cast is needed? Size is already int64_t, isn't it?

malfet commented on aten/src/ATen/native/TensorAdvancedIndexing.cpp:
```diff
@@ -677,13 +677,13 @@ AdvancedIndex::AdvancedIndex(const Tensor& src, TensorList indices_list) {
-  this->dims_before = dims_before;
-  this->dims_after = dims_after;
-  this->src = restride_src(src, dims_before, dims_indexed, replacement_shape);
+  this->dims_before = new_dims_before;
+  this->dims_after = new_dims_after;
```
You can remove `this->` as class members no longer shadowed by local variables

malfet commented on aten/src/ATen/native/TensorAdvancedIndexing.cpp:
```diff
@@ -647,14 +647,14 @@ static Tensor reshape_indexer(
 
 AdvancedIndex::AdvancedIndex(const Tensor& src, TensorList indices_list) {
```
Why not change `src` name to `src_` to avoid shadowing?

malfet commented on aten/src/ATen/native/TensorAdvancedIndexing.cpp:
```diff
@@ -1224,7 +1231,7 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
               IntArrayRef a, IntArrayRef b, int64_t dim) -> bool {
```
Why not make dim a `size_t`? Also, its name shadows method argument, isn't it?

malfet reviewed (changes requested):
I would avoid typecasting the loop argument, but rather fix others to be unsigned types

Also, if you want to get rid of variable shadowing, do it across all functions, not just where linter finds them

EricGriffith committed:
caffe2: Fix lint errors in native/TensorAdvancedIndexing.cpp (#150506)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/150506

See title

Test Plan: Sandcastle

Differential Revision: D72274536
